### PR TITLE
SNOW-998036: Add check if command text is set before executing

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Add check if command text is set before executing

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Review by sfc-gh-mhofman — APPROVED** [review]

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs [hunk 1]
```diff
@@ -34,6 +34,19 @@ public void TestCommandWithConnectionAndCommandText()
             Assert.AreEqual(commandText, command.CommandText);
         }
 
+        [Test]
+        public void TestCommandExecuteThrowsExceptionWhenCommandTextIsNotSet()
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs:38]: please add also a test for async command execution

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs (hunk 1 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs:40]: Added

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs [hunk 2]
```diff
@@ -34,6 +36,33 @@ public void TestCommandWithConnectionAndCommandText()
...
+        {
+            // Arrange
+            SnowflakeDbConnection conn = new SnowflakeDbConnection();
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs:43]: the conn variable is not used, and the test passes without it

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs [hunk 3]
```diff
@@ -34,6 +36,33 @@ public void TestCommandWithConnectionAndCommandText()
...
+
+            // Assert
+            Assert.AreEqual(thrown.Message, "Unable to execute command due to command text not being set");
+        }
+
+        [Test]
+        public void TestCommandExecuteAsyncThrowsExceptionWhenCommandTextIsNotSet()
+        {
+            // Arrange
+            SnowflakeDbConnection conn = new SnowflakeDbConnection();
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs:56]: the conn variable is not used, and the test passes without it

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs (hunk 2 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs]: Removed

File: Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs (hunk 3 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/UnitTests/SFDbCommandTest.cs]: Removed

**Review by sfc-gh-knozderko — APPROVED** [review]

**Pull request merged**